hooks/shotgun_get_valid_published_file.py

```diff
@@ -23,69 +24,68 @@
-        one of the viewer_extensions.
+        Returns the published file only if its path's extension matches one of
+        the valid_extensions.
 
         :param str entity_type: PublishedFile or TankPublishedFile.
         :param dict published_file: The published file.
         :returns: The published file with the right fields.
         :raises: TankError
         """
-        viewer_extensions = self.parent.get_setting("viewer_extensions")
-        if not viewer_extensions:
+        valid_extensions = self.parent.get_setting("valid_extensions")
+        if not valid_extensions:
             raise TankError(
-                "Sorry, viewer extensions must be provided."
+                "Sorry, valid extensions must be provided."
             )
-        sg_published_file = self.published_file(entity_type, published_file["id"])
-        path_on_disk = self.parent.get_path_on_disk(sg_published_file)
+        sg_published_file = self.parent.published_file(entity_type, published_file["id"])
+        path_on_disk = self.get_publish_path(sg_published_file)
         if path_on_disk:
-            for viewer_extension in viewer_extensions:
-                if path_on_disk.endswith(".%s" % viewer_extension):
+            for app_extension in valid_extensions:
+                if path_on_disk.endswith(".%s" % app_extension):
                     return sg_published_file
-        raise TankError("PublishedFile path %s does not match viewer extensions %s" % (
+        raise TankError("PublishedFile path %s does not match valid extensions %s" % (
             path_on_disk,
-            viewer_extensions
+            valid_extensions
         ))
 
     def resolve_multiple_files(self, published_file_type, published_files):
         """
         Decide which published file to return, or raise a TankError.
-        Return the first published file matching a viewer extension,
+        Return the first published file matching one of the valid_extensions,
         otherwise raise a TankError.
 
         :param str published_file_type: PublishedFile or TankPublishedFile.
         :param list published_files: The published files.
         :returns: The first published file with the right fields.
         :raises: TankError
         """
-        viewer_extensions = self.parent.get_setting("viewer_extensions")
-        if not viewer_extensions:
+
+        valid_extensions = self.parent.get_setting("valid_extensions")
+        if not valid_extensions:
             raise TankError(
-                "Sorry, viewer extensions must be provided."
+                "Sorry, valid extensions must be provided."
             )
         published_file_ids = [pf["id"] for pf in published_files]
-        published_files = self.sgtk.shotgun.find(
+        published_files = self.parent.shotgun.find(
             published_file_type,
             [["id", "in", published_file_ids]],
-            self._PUBLISHED_FILE_FIELDS
+            self.parent.PUBLISHED_FILE_FIELDS
```

Comment:
Oh dear... You're breaking encapsulation here,,, like with `self.parent.published_file`.

Reply:
And it's strange to have a `published_file ` method, but have use a shotgun.find method here. Is the `published_file` method needed?